workspace/php2/src/XSolveSecurityBundle/Controller/UserController.php

```diff
@@ -19,17 +20,70 @@ public function RestLoginAction(Request $request) {
       $autorizationManager = $this->get('x_solve_security.xsolve_autorization_manager');
       $authenticatedToken = $autorizationManager->getToken($username, $password);
 
-      if ($authenticatedToken) {
-	 $this->getUser()->assignNewToken();
-	 $this->getDoctrine()->getManager()->flush();
+      if (!$authenticatedToken) {
+
+	 throw new AccessDeniedException('unable to find token');
+      }
+      $this->getUser()->assignNewToken();
+      $this->getDoctrine()->getManager()->flush();
+
+      return (new JsonResponse())->setData(
+		      ['token' => $this->getUser()->getToken()->getToken()]);
+   }
+   public function showUserAction($id) {
+
+      $user = $this->getDoctrine()
+	      ->getRepository('XSolveSecurityBundle:User')
+	      ->find($id);
+
+      if (!$user) {
+
+	 throw $this->createNotFoundException(
+		 'No found user');
+      }
+
+      return (new JsonResponse())->setData(
+		      ['login' => $user->getUsername(),
+			'password' => $user->getPassword()]);
+   }
+   public function showAllUsersAction() {
+
+      $users = $this->getDoctrine()
+	      ->getRepository('XSolveSecurityBundle:User')
+	      ->findAll();
+
+      $tab = []; //tab contain id, usernames and passwords of all user in database
 
+      foreach ($users as &$user) {
 
-	 return (new JsonResponse())->setData(
-			 ['login' => $username,
-			  'password' => $username,
-		          'token' => $this->getUser()->getToken()->getToken()]);
+	 $element = ['id' => $user->getId(),
+	     'username' => $user->getUsername(), ];
+	 array_push($tab, $element);
       }
-      throw new AccessDeniedException('unable to find token');
+
+
+      $response = new jsonResponse();
+      $response->setData($tab);
+
+      return $response;
+   }
+
+   public function registerAction(Request $request) {
+      
+      $username = $request->request->get('login');
+      $password = $request->request->get('haslo');
+      $user = new User();

```

Comment:
Dobrym wyjściem było by gdyby konstruktor do Usera wyglądał tak:
   public function __construct($username = null, $password = null) {
      $this->username = $username;
      $this->password = $password;
   }
a kod tak z kontrolera tak:
  $username = $request->request->get('login');
  $password = $request->request->get('haslo');
  $user = new User($username, $password);
?